### ISSUE TITLE
[FG:InPlacePodVerticalScaling] Race condition setting pod resize status

### ISSUE THREAD

**Issue by tallclair** [kubernetes/kubernetes#125394]
The kubelet updates the pod status by first generating the pod status based on its internal representation, and then [merging](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/status/status_manager.go#L1049) that with the latest status from the API, based on whether the Kubelet should be the source of truth for those fields.

Currently, the PodResizeStatus is not synced with the API server status, which means that whatever the Kubelet thinks the resize status should be will overwrite whatever PodResizeStatus is currently stored in the API server. This can lead to a race condition where the Kubelet has not yet seen a resize update, and clobbers the `Proposed` resize status.

I'm not sure exactly what the right behavior is here, but it probably needs to be an intelligent state transition, or may require comparing resource versions.

/sig node
/kind bug
/cc @vinaykul @thockin

**Comment by AnishShah** [comment]
sig-node triage meeting:

- This will be fixed in https://github.com/kubernetes/kubernetes/pull/125958?

**Comment by AnishShah** [comment]
sig-node triage meeting:


/triage accepted
/priority important-longterm

**Comment by esotsal** [comment]
@tallclair this is an important PR for IPPVS to reach beta, correct ?  What is the status ? I have some time and could help on this PR if no one is actively working on this. Please let me know.

**Comment by esotsal** [comment]
I think this is fixed with https://github.com/kubernetes/kubernetes/pull/128269 , @tallclair do you agree?

**Comment by tallclair** [comment]
No, not by https://github.com/kubernetes/kubernetes/pull/128269. Our solution to this is that the API server should stop setting the `Proposed` resize status, and use ObservedGeneration instead.

/assign @natasha41575

**Comment by tallclair** [comment]
This will also be less relevant once we switch to using conditions for the resize status.
